File: avrodite-api/src/main/java/org/avrodite/api/CodecManager.java
package org.avrodite.api;

import java.lang.reflect.Type;

public interface CodecManager<S extends CodecStandard<?, ?, C, ?>, C extends Codec<?, ?, ?, S>> {
  <B, U extends Codec<B, ?, ?, S>> U getCodec(Class<B> beanType);
  <B, U extends Codec<B, ?, ?, S>> U getCodec(Type beanType);
}

File: avrodite-api/src/main/kotlin/org/avrodite/api/CodecManager.kt
package org.avrodite.api

import kotlin.reflect.KClass
import kotlin.reflect.KType

interface CodecManager<I:Input, O:Output, out C:Codec<*, I, O, Format<I, O, C, V>, V>, V : ValueCodec<*, I, O, Format<I, O, C, V>, C>> {
  fun <B : Any> getCodec(clazz: KClass<B>) : C
  fun <B : Any> getCodec(type: KType) : C
}
